sql/src/main/java/io/crate/operation/scalar/stats/PercentileFunction.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.operation.scalar.stats;
+
+import com.carrotsearch.hppc.DoubleArrayList;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import io.crate.metadata.FunctionIdent;
+import io.crate.metadata.FunctionInfo;
+import io.crate.metadata.Scalar;
+import io.crate.operation.Input;
+import io.crate.operation.scalar.ScalarFunctionModule;
+import io.crate.types.ArrayType;
+import io.crate.types.DataType;
+import io.crate.types.DataTypes;
+import org.apache.commons.math3.stat.descriptive.rank.Percentile;
+
+import java.util.Set;
+
+/**
+ * @see io.crate.operation.scalar.stats.PercentileFunction function estimates percentiles based on sample data.
```

Comment:
I think it would be good if this was less about implementation detail but more about what it does & API explanation for users of this function.

E.g. Just explain that it calculates a single percentile or multiple percentiles with two APIs:
percentile(fractionsArray, values) -> double[]
percentile(fraction, values) -> double